### PR TITLE
Query: Don't bind with subqueries if subquery requires client projection

### PR DESCRIPTION
**Pull request by smitpatel**
@anpete

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Relational/Query/RelationalQueryModelVisitor.cs [hunk 1]
```diff
@@ -1571,13 +1571,16 @@ public override Expression BindMemberToValueBuffer(MemberExpression memberExpres
                     RelationalQueryModelVisitor subQueryModelVisitor;
                     if (_subQueryModelVisitorsBySource.TryGetValue(querySource, out subQueryModelVisitor))
                     {
-                        selectExpression = subQueryModelVisitor.Queries.SingleOrDefault();
+                        if (!subQueryModelVisitor.RequiresClientProjection)
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/RelationalQueryModelVisitor.cs]: Is this requires client anything?

Comment (smitpatel) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/RelationalQueryModelVisitor.cs]: ClientEverything.

**Comment by smitpatel** [comment]
Recalling. Complex issue it is!

**Comment by anpete** [comment]
@smitpatel Soon, Yoda of LINQ you will be!

**Pull request closed without merging**